### PR TITLE
29 3.6 counter

### PR DESCRIPTION
**Pull request by frederike-ramin**
Add a counter of free and occupied places for an event.

### CODE REVIEW

**Review by Baschdl — DISMISSED** [review]
Looks good. Please wait for approval of (full-time) developer ;)

File: db/migrate/20161103101916_create_applications.rb
Comment (ekrebs5) [inline db/migrate/20161103101916_create_applications.rb]: (this is a question about our intern conventions, not about the code itself. This code should work with the new and the old version, I tested both on my system)
This migration was written when "event" was still called "workshop". Personally I think it should stay workshop in this method, otherwise we could rename all other migration methods and remove the "workshop->event" migration (This would however probably mean that everyone has to reset their database at least once). Do we have conventions/rules for this?

Comment (Baschdl) [inline db/migrate/20161103101916_create_applications.rb]: Yes, that's right. This renaming will probably crash the migration process. I'll talk with @bjrne from #62

Comment (bjrne) [inline db/migrate/20161103101916_create_applications.rb]: You should never touch old migration files (afaik). This is the old file that created the workshop column that later was renamed to event. I don't know what happens if you mess with this file and we need to execute it again. Please remove this change.

**Review by bjrne — DISMISSED** [review]
Please remove the changes to the old migration file.

File: db/migrate/20161103101916_create_applications.rb
Comment (bjrne) [inline db/migrate/20161103101916_create_applications.rb]: @ekrebs5 I think the reason it both worked for you is that Ruby is so smart, it doesn't repeat migrations it already applied. Because the filename didn't change, this change was never tested when you just migrated your database. To really check this, you'd have to manually reverse all applied migrations until then, an migrate everything again. I am pretty sure the renaming would fail with this change, because there would be no more ":workshop"-column.

Comment (bjrne) [inline db/migrate/20161103101916_create_applications.rb]: Look at [this](http://edgeguides.rubyonrails.org/active_record_migrations.html#changing-existing-migrations) for more details: 
"In general, editing existing migrations is not a good idea. You will be creating extra work for yourself and your co-workers and cause major headaches if the existing version of the migration has already been run on production machines."

File: db/schema.rb
Comment (bjrne) [inline db/schema.rb]: Why do you have changes in this file when you didn't create any migrations? Have a loot at [this page](http://edgeguides.rubyonrails.org/active_record_migrations.html#schema-dumping-and-you) to get a better understanding of schema files if you didn't just forget to commit the migration file 😀

File: app/views/events/show.html.erb
Comment (bjrne) [inline app/views/events/show.html.erb]: Is this whitespace change necessary?

File: app/controllers/application_letters_controller.rb
Comment (bjrne) [inline app/controllers/application_letters_controller.rb]: I think this was removed a couple PRs earlier. I don't know your code that well, I just want to make sure this is not a merge error. Please give it a second look 🙂

Comment (aloeser) [inline app/controllers/application_letters_controller.rb]: do you know why it was removed? An application letter without an user seems not really useful to me

Comment (frederike-ramin) [inline app/controllers/application_letters_controller.rb]: My guess is, that you shoulndn't be able to create a letter with a manually set user. instead, in the create function, the user_id should be set by the backend itself.

Comment (cmfcmf) [inline app/controllers/application_letters_controller.rb]: > My guess is, that you shoulndn't be able to create a letter with a manually set user. instead, in the create function, the user_id should be set by the backend itself.

Exactly. 😄

Comment (aloeser) [inline app/controllers/application_letters_controller.rb]: @cmfcmf are there already tests for this? if yes, we also broke the tests.
If no, maybe your group should add some, since this seems like an important security feature :)

File: app/views/events/show.html.erb
Comment (aloeser) [inline app/views/events/show.html.erb]: fixed

**Review by ekrebs5 — APPROVED** [review]
I think this is now ok :)

File: spec/features/events_spec.rb
Comment (querenker) [inline spec/features/events_spec.rb]: Maybe a loop could improve the readability of this test

**Review by querenker — APPROVED** [review]

**Pull request merged**
